src/modm/driver/encoder/encoder.lb

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2019, Thomas Sommer
+#
+# This file is part of the modm project.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+# -----------------------------------------------------------------------------
+
+def init(module):
+    module.name = ":driver:encoder"
+    module.description = """
+Software graycode ported from C-Code of Peter Dannegger:
+https://www.mikrocontroller.net/articles/Drehgeber.
```

Comment:
> Following an example i've seen in another modm-module, i've put the credits to Peter Dannegger's ANSI-C foundation into encoder.lb ... but this may not be sufficient.

I think it's fine, we've haven't copied the code literally and we're even adapting/improving it.